### PR TITLE
docs: Update ConfigurationModifier property documentation

### PR DESCRIPTION
**Pull request by jskeet**
(This is initially just one property, but once we've tweaked it until we're happy, I'll roll it out to all the equivalent properties.)

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadJsonOptions.cs
Comment (Rishabh-V) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadJsonOptions.cs:86]: It looks good. 👍🏼 This conveys the message and helps me use this property correctly. 
I think we can break the sentence for even better clarity. Something like:
```Properties within this type are used to prepare the request before the delegate is called. Therefore, the delegate should generally not modify .........```
But it is fine without that change as well.

**Review by Rishabh-V — APPROVED** [review]
LGTM.

Also shared my thoughts inline.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadJsonOptions.cs
Comment (amanda-tarafa) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadJsonOptions.cs:86]: I think something along the lines of preference would be clearer:

"Prefer the properties on this type over this modifier to prepare the request. Only use this modifier to configure aspects for which there are no properties available. This modifier is applied to the request after all properties on this type have been applied."

**Review by amanda-tarafa — COMMENTED** [review]
I've proposed an alternative, but happy as is as well.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadJsonOptions.cs
Comment (jskeet) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadJsonOptions.cs:86]: @amanda-tarafa: Works for me. So that's just changing the first sentence in the remarks, and I'm okay to keep the one about "The delegate is only called once [...]"?

Comment (amanda-tarafa) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadJsonOptions.cs:86]: Yes, exactly. And also yes, keep the one about the delegate being called only once.

**Comment by jskeet** [comment]
I've now applied Amanda's suggested change, and then applied the change to all jobs. PTAL.

**Review by amanda-tarafa — APPROVED** [review]
Thanks!

**Comment by jskeet** [comment]
(Will wait until the morning to merge, in case the Lex team has further comments.)

**Comment by Rishabh-V** [comment]
> (Will wait until the morning to merge, in case the Lex team has further comments.)

This looks good to me. Thanks.

**Pull request merged**
